### PR TITLE
IBM IoT sample

### PR DESCRIPTION
**Pull request by andrey4latyshev**
A sample to test IBM device connection. It is not approved by Sample Workgroup yet so marked as  WIP.

### CODE REVIEW

**Comment by anjgola** [comment]
@mkitez  interesting case to observe here. Checks passed for missing sample.json, I guess it could be that the sample does not require sample.json but I think we should fail the CI and let it be an explicit override by maintainer to merge this sample. Something to think about, doesn't affect this MR since Andrey plans to enter a sample.json and had marked this request as WIP.

**Comment by JoeOster** [comment]
@andrey4latyshev - Please take a look at and resolve the CI error, error message is below
You have called ADD_LIBRARY for library iotp without any source files. This typically indicates a problem with your CMakeLists.txt file
CMake Error: Cannot determine link language for target "iotp".
CMake Error: CMake can not determine linker language for target: iotp

File: Tools/IoTConnectionTools/ibm-device/License.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Copyright 2019 Intel Corporation
```
Comment (JoeOster) [inline Tools/IoTConnectionTools/ibm-device/License.txt:1]: Remove date from License.txt

File: Tools/IoTConnectionTools/ibm-device/sample.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "IBM Device",
+  "categories": ["Toolkit/Intel® oneAPI IoT Toolkit/IoT Connection Tools"],
```
Comment (JoeOster) [inline Tools/IoTConnectionTools/ibm-device/sample.json:3]: Need to add the guid, per the move document

**Review by mkitez — APPROVED** [review]
Approving to trigger CI

File: Tools/IoTConnectionTools/ibm-device/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+Configure IoT device on [IBM Watson IoT Platform Page](https://ibm-watson-iot.github.io/iot-c/device/).
+
+### Application Parameters
+
+The samples uses path to configuration file as a parameter.
+Download the configuration file with all the credentials according to [instructions](https://ibm-watson-iot.github.io/iot-c/device/).
+
+### Example of Output
+
+TBD
```
Comment (JoeOster) [inline Tools/IoTConnectionTools/ibm-device/README.md:70]: Please resolve the TBD

File: Tools/IoTConnectionTools/ibm-device/cpp/deviceSample.c [hunk 4]
```diff
@@ -0,0 +1,244 @@
+/*******************************************************************************
```
Comment (JoeOster) [inline Tools/IoTConnectionTools/ibm-device/cpp/deviceSample.c:1]: As far as I can tell, this was not previously disclosed as a dependency. Please inform Swapna that this sample will require OSPDT approval for deviceSimple

Comment (andrey4latyshev) [reply Tools/IoTConnectionTools/ibm-device/cpp/deviceSample.c:1]: > As far as I can tell, this was not previously disclosed as a dependency. Please inform Swapna that this sample will require OSPDT approval for deviceSimple

The IBM Watson IoT Platform C Client SDK is approved as 'Conditional' (Slide 11 of Open Source PDT v7.0 - oneAPI samples Beta09.pptx). This sample (deviceSample) is a part of the mentioned approved library.

Comment (JoeOster) [reply Tools/IoTConnectionTools/ibm-device/cpp/deviceSample.c:1]: Thank you for the clarification

File: Tools/IoTConnectionTools/ibm-device/README.md (hunk 3 above)
Comment (andrey4latyshev) [reply Tools/IoTConnectionTools/ibm-device/README.md:70]: This TBD can be defined after additional research and after resolving some other issues.

File: Tools/IoTConnectionTools/ibm-device/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+Perform the following steps:
+1. Run in the terminal:
+    ```
+    cd $ENV{HOME}
+    git clone https://github.com/ibm-watson-iot/iot-c.git
+    cd iot-c
+    make
+    sudo make -C paho.mqtt.c install
+    sudo make install
+    ```
```
Comment (anjgola) [inline Tools/IoTConnectionTools/ibm-device/README.md:41]: Are these instructions applicable anymore?

Comment (JoeOster) [reply Tools/IoTConnectionTools/ibm-device/README.md:41]: > Are these instructions applicable anymore?

@andrey4latyshev - can you take a look at this

Comment (andrey4latyshev) [reply Tools/IoTConnectionTools/ibm-device/README.md:41]: Yes, these instructions help to build prerequisites.

Comment (anjgola) [reply Tools/IoTConnectionTools/ibm-device/README.md:41]: Why ask them to git clone the sample from that repo when it is present in this repo?

**Comment by anjgola** [comment]
Pending on Swapna's approval

File: Tools/IoTConnectionTools/ibm-device/README.md (hunk 5 above)
Comment (andrey4latyshev) [reply Tools/IoTConnectionTools/ibm-device/README.md:41]: > Why ask them to git clone the sample from that repo when it is present in this repo?

This step is needed to clone the dependency library IBM Watson IoT Platform  from git.

**Comment by srdontha** [comment]
We have OS PDT and legal approval for the IBM Watson IoT Platform SDK. Ok from my side. Thanks.

**Pull request merged**